logstash/lib/logstash/outputs/loki.rb

```diff
@@ -0,0 +1,279 @@
+# encoding: utf-8
+require "logstash/outputs/base"
+require "logstash/namespace"
+require 'net/http'
+require 'concurrent-edge'
+require 'time'
+require 'uri'
+require 'json'
+
+# An example output that does nothing.
+class LogStash::Outputs::Loki < LogStash::Outputs::Base
+  require 'logstash/outputs/loki/batch'
+  require 'logstash/outputs/loki/entry'
+
+  config_name "loki"
+
+  ## 'A single instance of the Output will be shared among the pipeline worker threads'
+  concurrency :single
+
+  ## 'Loki URL'
+  config :url, :validate => :string,  :required => true
+
+  ## 'BasicAuth credentials'
+  config :username, :validate => :string, :required => false
+  config :password, :validate => :string, secret: true, :required => false
+
+  ## 'Client certificate'
+  config :cert, :validate => :path, :required => false
+  config :key, :validate => :path, :required => false
+
+  ## 'TLS'
+  config :ca_cert, :validate => :path, :required => false
+
+  ## 'Loki Tenant ID'
+  config :tenant_id, :validate => :string, :default => "fake", :required => false
+
+  ## 'Maximum batch size to accrue before pushing to Loki. Defaults to 102400 bytes'
+  config :batch_size, :validate => :number, :default => 102400, :required => false
+
+  ## 'Interval in seconds to wait before pushing a batch of records to Loki. Defaults to 1 second'
+  config :batch_wait, :validate => :number, :default => 1, :required => false
+
+  ## 'Array of label names to include in all logstreams'
+  config :include_labels, :validate => :array, :default => [], :required => true 
```

Comment:
@cyriltovena Should we add this parameter? or just exclude labels is enough?

Reply:
Is include_labels used anywhere else in the code? Not seeing it referenced. Would expect it being used in e.g. `extract_labels`

Reply:
Yeah, I did not add anywhere in the code still. Just wanted to check if we want this parameter

Reply:
yup, looks like a good addition

Reply:
👍

Reply:
I'm confused we have both include and exclude labels ?

Reply:
Yeah, kind of duplicate. User can choose to have all labels but exclude some or explicitly define the labels to include. This is to just make sure user doesn't include high cardinality value.

Reply:
the discussion we had here https://grafana.slack.com/archives/CTJ4EN6MU/p1585662854039400

Reply:
then include_labels might be enough.

Reply:
This is kind of tough to decide. include_labels will work but if the event has lot of labels and the user wants all labels they would have to manually include it

Reply:
Yeah I prefer the pain to add a lot of labels than the ease :) Trust me they will thanks us !